internal/remote/sources.go

```diff
@@ -0,0 +1,133 @@
+package remote
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	core "github.com/speakeasy-api/speakeasy-core/auth"
+	"github.com/speakeasy-api/speakeasy/internal/sdk"
+
+	"github.com/speakeasy-api/speakeasy-client-sdk-go/v3/pkg/models/operations"
+	"github.com/speakeasy-api/speakeasy-client-sdk-go/v3/pkg/models/shared"
+)
+
+// RecentGeneration represents a recent generation of a target in the workspace.
+// The source of this data is our CLI event stream, which is updated every time
+// a target is generated.
+type RecentGeneration struct {
+	CreatedAt            time.Time
+	ID                   string
+	TargetName           string
+	Target               string
+	SourceNamespace      string
+	SourceRevisionDigest string
+	Success              bool
+	Published            bool
+
+	// May not be set
+	GitRepoOrg *string
+	GitRepo    *string
+
+	// gen.yaml
+	GenerateConfig *string
+}
+
+const (
+	// The event stream contains multiple events for the same namespace, so we want to
+	// break execution once we've seen a minimum, arbitrary number of unique namespaces.
+	recentGenerationsToShow int = 5
+)
+
+// GetRecentWorkspaceGenerations returns the most recent generations of targets in a workspace
+// This is based on the CLi event stream, which is updated on every CLI interaction.
+func GetRecentWorkspaceGenerations(ctx context.Context) ([]RecentGeneration, error) {
+	workspaceId, err := core.GetWorkspaceIDFromContext(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	speakeasyClient, err := sdk.InitSDK()
+	if err != nil {
+		return nil, err
+	}
+
+	// The event stream is limited to the most recent 250 events
+	res, err := speakeasyClient.Events.SearchWorkspaceEvents(ctx, operations.SearchWorkspaceEventsRequest{
```

Comment:
Why not use `GetRecentTargets`?

Reply:
realized I left the same comment. I'm ok not using GetRecentTargets but I do think we need some priority rankings on targets we want to surface and targets we don't